advanced-data-types.md

```diff
@@ -0,0 +1,195 @@
+
+    public string Name
+    {
+        get => _Name;
+        set
+        {
+            ValidateName(value);
+            _Name = value;
+        }
+    }
+}
+```
+
+This is quite neat as it makes the public API of your model safe, while preserving their persistability. Of note is the `MapTo` attribute applied to `_Name` - it is not strictly necessary, I just added it to avoid having ugly column names in the database. You can use it or not use it - it's totally up to you. One thing to note when utilizing this technique is that Realm is completely unaware of the relationship between `Name` and `_Name`. This has two implications - 1) notifications will be emitted for `_Name` only and 2) you can't use LINQ queries to filter `Person` objects by name. Let's see how we can mitigate both:
+
+For notifications, we can override `OnPropertyChanged` and raise a notification for `Name` whenever `_Name` changes:
+
+```csharp
+class Person : RealmObject
+{
+    protected override void OnPropertyChanged(string propertyName)
+    {
+        base.OnPropertyChanged(propertyName);
+
+        if (propertyName == nameof(_Name))
+        {
+            RaisePropertyChanged(nameof(Name));
+        }
+    }
+}
+```
+
+The code is fairly straightforward - `OnPropertyChanged` will be invoked whenever any property on the object changes and we just re-raise it for the related `Name` property. Note that, as an optimization, `OnPropertyChanged` will only be invoked if there are subscribers to the `PropertyChanged` event. So if you're testing this out and don't see the code get executed, make sure you added a subscriber first.
+
+The situation with queries is slightly harder to work around. The main issue is that, because the property is private, you can't use it in a LINQ query - e.g. `realm.All<Person>().Where(p => p._Name == "Peter")` will result in a compile-time error. On the other hand, because Realm doesn't know that `Name` is tied to `_Name`, you can't use `p.Name == "Peter"` either. You can still use the string-based queries though, just remember to use the name that Realm knows about - i.e. the string argument of `MapTo` if you remapped the property name or the internal property (`_Name`) if you didn't:
+
+```csharp
+// _Name is mapped to 'Name' which is what we use here
+var peters = realm.All<Person>().Filter("Name == 'Peter'");
+```
+
+## Using unpersistable data types
+
+Realm has a wide variety of supported data types - most primitive types in the BCL, as  well as advanced collections, such as sets and dictionaries. But sometimes, you'll come across a data type that Realm can't store yet - the most obvious example being enums. In such cases, you can build on top of the previous technique to expose enum properties in your models and have them be persisted as one of the supported data types:
+
+```csharp
+enum TransactionState
+{
+    Pending,
+    Settled,
+    Error
+}
+
+class Transaction : RealmObject
+{
+    private string _State { get; set; }
```

Comment:
Is `string` actually our recommendation for `enum`? Wouldn't `int` be more intuitive to use?

Reply:
edit: I found the answer partially in the paragraph below but will leave the rest here out of interest.

Reply:
Yeah, `string` is the recommended way to store enums in databases.